crates/router/src/core/payments/operations/payment_response.rs

```
@@ -2544,47 +2544,20 @@ impl<F: Clone> PostUpdateTracker<F, PaymentConfirmData<F>, types::PaymentsAuthor
-                | common_enums::AttemptStatus::Charged
-                | common_enums::AttemptStatus::Authorizing
-                | common_enums::AttemptStatus::CodInitiated
-                | common_enums::AttemptStatus::PartialCharged
-                | common_enums::AttemptStatus::PartialChargedAndChargeable
-                | common_enums::AttemptStatus::CaptureInitiated
-                | common_enums::AttemptStatus::PaymentMethodAwaited
-                | common_enums::AttemptStatus::ConfirmationAwaited
-                | common_enums::AttemptStatus::DeviceDataCollectionPending
-                | common_enums::AttemptStatus::IntegrityFailure => {
-                    let pm_update_status = enums::PaymentMethodStatus::Active;
-
-                    // payment_methods microservice call
-                    payment_methods::update_payment_method_status_internal(
-                        state,
-                        key_store,
-                        storage_scheme,
-                        pm_update_status,
-                        payment_method.get_id(),
-                    )
-                    .await
-                    .change_context(errors::ApiErrorResponse::InternalServerError)
-                    .attach_printable("Failed to update payment method status")?;
-                }
+            let pm_update_status = enums::PaymentMethodStatus::from(&attempt_status);
+
+            if pm_update_status == enums::PaymentMethodStatus::Active {
+                // payment_methods microservice call
+                payment_methods::update_payment_method_status_internal(
+                    state,
+                    key_store,
+                    storage_scheme,
+                    pm_update_status,
+                    payment_method.get_id(),
+                )
+                .await
+                .change_context(errors::ApiErrorResponse::InternalServerError)
+                .attach_printable("Failed to update payment method status")?;
```

do we want to propagate the error if we failed to updated payment method? shouldn't this be silent error?